toolboxes/core/gpu/cuNDArray_blas.cu

```diff
@@ -1,312 +1,509 @@
-#include "cuNDArray_blas.h"
-#include "complext.h"
 #include "GadgetronCuException.h"
+#include "complext.h"
+#include "cuNDArray_blas.h"
 #include "cudaDeviceManager.h"
 
 #include <cublas_v2.h>
 
-namespace Gadgetron{
-
-#define CUBLAS_CALL(fun) {cublasStatus_t err = fun; if (err != CUBLAS_STATUS_SUCCESS) {throw cuda_error(gadgetron_getCublasErrorString(err));}}
-
-  //NRM2
-  //
+namespace Gadgetron {
 
-  template<class T> cublasStatus_t cublas_axpy(cublasHandle_t hndl, int n, const T* a , const T* x , int incx,  T* y, int incy);
-  template<class T> cublasStatus_t cublas_dot(cublasHandle_t, int, const T*, int, const  T*, int, T*, bool cc = true);
-  template<class T> cublasStatus_t cublas_nrm2(cublasHandle_t, int, const T*, int, typename realType<T>::Type *result);
-  template<class T> cublasStatus_t cublas_amax(cublasHandle_t handle, int n,const T *x, int incx, int *result);
-  template<class T> cublasStatus_t cublas_amin(cublasHandle_t handle, int n,const T *x, int incx, int *result);
-  template<class T> cublasStatus_t cublas_asum(cublasHandle_t handle, int n,const T *x, int incx, typename realType<T>::Type *result);
+#define CUBLAS_CALL(fun)                                                                                               \
+    {                                                                                                                  \
+        cublasStatus_t err = fun;                                                                                      \
+        if (err != CUBLAS_STATUS_SUCCESS) {                                                                            \
+            throw cuda_error(gadgetron_getCublasErrorString(err));                                                     \
+        }                                                                                                              \
+    }
 
-  template<> cublasStatus_t cublas_nrm2<float>(cublasHandle_t hndl, int n, const float*  x, int inc, float* res){
-    return cublasSnrm2(hndl,n,x,inc,res);
-  }
+template <class T>
```

Comment:
The formatting changes here seem arbitrary. Looks like your editor (?) is trying to limit the max width to some number of characters....I know that it is referred in some projects but it is not enforced here and in this case, it just makes things less readable.

Reply:
@ahsanjav the arbitrary formatting changes remain here. I think that should be backed out. Like here there are sometimes line breaks after `template <class T>` and sometimes not. We are not enforcing a specific line length and here it really hurts readability.